plugins/single_plugins/oswitch.cpp

```diff
@@ -33,8 +53,50 @@ class wayfire_oswitch : public wf::plugin_interface_t
         return os[target_index];
     }
 
+    wf::output_t *get_output_in_direction(int dir_x, int dir_y)
+    {
+        auto current_output = wf::get_core().seat->get_active_output();
+        if (!current_output)
+        {
+            return nullptr;
+        }
+
+        auto current_geo = current_output->get_layout_geometry();
+
+        wf::output_t *best_output = nullptr;
+        double best_score = -INFINITY;
+
+        for (auto& output : wf::get_core().output_layout->get_outputs())
+        {
+            if (output == current_output)
+            {
+                continue;
+            }
+
+            auto geo  = output->get_layout_geometry();
```

Comment:
I am wondering about the logic here. You select the one with the biggest distance from the center, right? So say we have three monitors A, B, C in a row, left to right. Now if I am on A and press the binding to go to the right, I expect B to be focused. But your current logic would select C (if I understand correctly? As C would give the highest score with the longest distance). I am not sure what the best way to solve this is, maybe try to look for closest outputs to the right edge of the output (when going to the right?)